smartling-files-api/src/main/java/com/smartling/api/files/v2/FilesApiFactory.java

```diff
@@ -28,13 +31,18 @@ protected Class<FilesApi> getApiClass()
     @Override
     public FilesApi buildApi(final AuthorizationRequestFilter authFilter, ClientConfiguration config)
     {
+        ResteasyProviderFactory resteasyProviderFactory = config.getResteasyProviderFactory();
+        if (null == resteasyProviderFactory)
+            resteasyProviderFactory = new LocalResteasyProviderFactory(ResteasyProviderFactory.getInstance());
```

Comment:
It is a bit changed initialization of providerFactory, if you debug AbstractApiFactory.buildApi with passed null as providerFactory you will find the following code in ResteasyClientBuilder
```
protected ResteasyProviderFactory getProviderFactory() {
        if (this.providerFactory == null) {
            this.providerFactory = new LocalResteasyProviderFactory(new ResteasyProviderFactory());
            if (!(Boolean)AccessController.doPrivileged(() -> {
                String propertyValue = ResteasyConfigFactory.getConfig().getValue("resteasy.client.providers.annotations.disabled", SOURCE.SYSTEM);
                return propertyValue == null ? Boolean.FALSE : Boolean.valueOf(propertyValue.isEmpty() ? "true" : propertyValue);
            }) && ResteasyProviderFactory.peekInstance() != null) {
                this.providerFactory.initializeClientProviders(ResteasyProviderFactory.getInstance());
            }

            RegisterBuiltin.register(this.providerFactory);
        }

        return this.providerFactory;
    }
```
Are you sure it is ok as well?

Reply:
changed to use default factory initialization